Therify/directory
# feat: handle inactive and expired plans

# Description
Adds a hook to all navigation page components that watches user plans and checks:
- status is active or trialing
- Start date is in the past
- End date is in the future
If any of those checks fail, the user will be redirected to the appropriate page.
Members go to `/members/account/expired-plan`
Providers go to `/providers/account/billing`
 If start date is in the future, all users go to `/access-countdown`

# Closes issue(s)
[Check plan expiration when provisioning access to app](https://trello.com/c/b1fpdEvK)

# How to test / repro
sign in as practice owner.
Go to stripe customer portal and cancel plan
see access get revoked

# Screenshots
### Plan status not active
<img width="1787" alt="image" src="https://user-images.githubusercontent.com/25045075/223915788-015d2ef6-e5e9-4e4f-9a9b-c4560364a597.png">

### Plan expired
<img width="1391" alt="image" src="https://user-images.githubusercontent.com/25045075/223919201-90d3e642-513f-4aaf-bb34-6b89b8c00783.png">

### Member plan expires/not active
<img width="1394" alt="image" src="https://user-images.githubusercontent.com/25045075/223919350-d18d6d02-bb30-47b2-87bd-3b05352a33b0.png">

### Plan not active yet
<img width="997" alt="image" src="https://user-images.githubusercontent.com/25045075/223918980-b11cf41b-af47-456e-b71f-7a467db0913c.png">


# Changes include


# Checklist


# Other comments
